src/user.ts

```diff
@@ -75,6 +75,14 @@ async function retrieveUserFields(logger: Logger, username: string): Promise<Use
     authInfo: await AuthInfo.create({ username }),
   });
 
+  if (sfdc.matchesAccessToken(username)) {
```

Comment:
should resolving the username like this be used in creating the Connection above?

Reply:
I don't understand Auth well enough to know, so I'm limiting the change to fixing the bug.

It definitely works OK to get the Connection via accessToken.

Reply:
No we must use the access token for the connection if it's passed in.  If we used the username it would fail because there's not associated auth file.